### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by romulogcerqueira**
Refactored codeSonarData() method to avoid corner cases

### CODE REVIEW

File: src/MultibeamSonar.cpp [hunk 1]
```diff
@@ -34,37 +34,33 @@ std::vector<float> MultibeamSonar::codeSonarData(const cv::Mat3f& cv_image) {
...
-
-			while (col_end == -1) {
-				cv::minMaxIdx(shader[2].col(++i), NULL, &max2);
-				i < middle_img ? max2 = -max2 : max2 += 0;
-				int id_curr = a * (max2 + 1);
-				if (id_curr != id_beam)
-					col_end = --i;
+			double max_angle;
+			cv::minMaxIdx(shader[2].col(i), NULL, &max_angle);
+			i < middle_img ? max_angle = (1 - max_angle) * 0.5 : max_angle = (1 + max_angle) * 0.5;
```
Comment (doudou) [inline src/MultibeamSonar.cpp]: Don't use the ternary operator for complex operations. `if() ... else()` is really more readable

**Comment by doudou** [comment]
One thing I don't understand in the code. If you don't use a perspective camera model, all pixels in a column would all have the same angle... So why do you need to compute the max angle of a column at all ? The mapping from column to beam ID should be constant.

**Comment by romulogcerqueira** [comment]
Good point,@doudou.

Through shader image, all pixels in a column present the same angle value, however I need to check it for two reasons:
* I don't know the angle value *a priori*;
* I only have the angle value if there is some object present in that column. This solution on shader saves processing time.

File: src/MultibeamSonar.cpp (hunk 1 above)
Comment (romulogcerqueira) [inline src/MultibeamSonar.cpp]: Ok. Fixed!

**Comment by doudou** [comment]
> I don't know the angle value a priori;

I don't understand why not

**Pull request closed without merging**